plugin/pkg/kubectl/client/plugin_client.go

```diff
@@ -0,0 +1,192 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package client
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"time"
+
+	restclient "k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+
+	"github.com/kubernetes-incubator/service-catalog/pkg/client/clientset_generated/clientset"
+)
+
+//PluginClient is a client for interacting with the service catalog
+//via generated clientset interface
+type PluginClient struct {
+	ScClient clientset.Interface
+	Config   *restclient.Config
+}
+
+// NewClient uses the KUBECONFIG environment variable to create a new client
+// based on an existing configuration
+func NewClient() (*PluginClient, error) {
+	// resolve kubeconfig location, prioritizing the --config global flag,
+	// then the value of the KUBECONFIG env var (if any), and defaulting
+	// to ~/.kube/config as a last resort.
+	home := os.Getenv("HOME")
+	kubeconfig := home + "/.kube/config"
+
+	kubeconfigEnv := os.Getenv("KUBECONFIG")
```

Comment:
This logic was already processed in `kubectl`, you should just rely on the plugin-specific env vars (`KUBECTL_PLUGINS_*`) you get.

Reply:
I was actually unaware of that, thanks for the heads up. I'd like to try to limit discussion on the plugin client to my PR [here](https://github.com/kubernetes/kubectl/pull/182) as the client here in service-catalog is strictly temprorary.. Once the PR to kubernetes/kubectl gets merged I'll be switching this plugin to use the client over there.